pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -492,7 +492,8 @@ def _set_parallel_devices_and_init_accelerator(self) -> None:
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
                     f" got `devices={self._devices_flag}` instead."
                 )
-
+        if not self.accelerator.is_available():
+            raise MisconfigurationException(f"{self.accelerator.__class__.__qualname__} can not run on this hardware.")
```

Comment:
Could we change the exception to X devices is not available? I think users are more familiar with that